lightningd/peer_control.c

```diff
@@ -619,6 +620,16 @@ static void json_add_channel(struct lightningd *ld,
 	json_add_string(response, "channel_id",
 			type_to_string(tmpctx, struct channel_id, &cid));
 	json_add_txid(response, "funding_txid", &channel->funding_txid);
+
+	if (channel->shutdown_scriptpubkey[LOCAL]) {
+		json_add_string(response, "close_to_addr",
+				encode_scriptpubkey_to_addr(tmpctx,
+					get_chainparams(ld),
+					channel->shutdown_scriptpubkey[LOCAL]));
```

Comment:
Should we handle encode_scriptpubkey_to_addr returning NULL?  The classic would be not to add the field if it's NULL...

Reply:
good idea. updated.